openedx/xblock-skill-tagging
# [BB-6857] Add tagging mixins

## Description
This PR adds a new mixin to xblock - `SkillTaggingMixin` that allows users to fetch and verify skill tags associated with xblocks. It also introduces a new field `has_verified_tags` which indicates whether the xblock contains skill tags that are verified.

For fetching the skill tags, it uses the taxonomy API. The taxonomy API base url should be set via the settings module using `TAXONOMY_API_BASE_URL` field.

## Additional Information:
OpenCraft's Jira task: [BB-6857](https://tasks.opencraft.com/browse/BB-6857)

## Testing Instructions
1. Clone this repo in `<devstack_base_dir>/src/ and checkout this branch.
2. Open lms shell: `make lms-shell`.
3. Install this repo using `pip install -e /src/xblock-skill-tagging`
4. Create a `private.py` file in `lms/envs/` and add the following settings:
```
from .common import XBLOCK_MIXINS
XBLOCK_MIXINS += ('skill_tagging.skill_tagging_mixin.SkillTaggingMixin',)
```
5. Add the setting `TAXONOMY_API_BASE_URL` to the endpoint where taxonomy is installed (like discovery).
6. Restart lms service.
7. Open lms shell and get any XBlock object.
8. Verify that the new attributes are added to the XBlock.
